src/hooks/use-passcode.hook.ts

```diff
@@ -0,0 +1,28 @@
+import { useCallback, useState } from 'react';
+import { isPinOrFingerprintSet } from 'react-native-device-info';
+import { useDispatch } from 'react-redux';
+
+import { resetApplicationAction } from '../store/root-state.actions';
+import { useAppStateStatus } from './use-app-state-status.hook';
+
+export const usePasscode = () => {
+  const dispatch = useDispatch();
+  const [isPasscode, setIsPasscode] = useState<null | boolean>(null);
+
+  const isPinOrFingerPrintHandler = useCallback((isPinOrFingerprintSet: boolean) => {
```

Comment:
`isPinState` and `isPinOrFingerPrintHandler` is used only once. omit this variables and just put their code into `useAppStateStatus` as you will do this with `useEffect`